ui/desktop/app/routes/scopes/scope/projects/targets/target.js

```diff
@@ -23,19 +23,44 @@ export default class ScopesScopeProjectsTargetsTargetRoute extends Route {
    * Load a target
    * @param {object} params
    * @param {string} params.target_id
-   * @param {bool} params.isConnecting
    * @return {TargetModel}
    */
-  async model({ target_id, isConnecting }) {
-    const target = await this.store.findRecord('target', target_id, {
+  model({ target_id }) {
+    return this.store.findRecord('target', target_id, {
       reload: true,
     });
+  }
 
-    if (isConnecting) {
-      await this.preConnect(target);
-    }
+  async afterModel(model, transition) {
+    const { isConnecting } = transition.to.queryParams;
+
+    if (isConnecting && model.address) {
+      await this.connect(model);
+    } else {
+      const hostSets = await Promise.all(
```

Comment:
Do you need to await this here?

Reply:
Yes. This tells the top level promise to not resolve until all of the `this.store.findRecord('host-set', host_source_id)` calls resolve. Similar to line 54... expect we don't actually need the `async/await` on the individual `findRecord` call. So I have removed those.